Sources/core/objects/Projectiles.cpp

```diff
@@ -0,0 +1,254 @@
+#include "Projectiles.h"
+#include "../Helpers.h"
+
+
+Projectile::Projectile(size_t id) : IMovable(id)
+{
+    SetPassable(D_ALL, Passable::AIR);
+    tickSpeed = 0.25;
+    damage = 0;
+    SetSprite("icons/projectiles.dmi");
+    v_level = 5;
+}
+void Projectile::Process()
+{
+    if(set_target_ == 0)
+    {
+       return;
+    }
+    int z = current_step_;
+    Dir step = VDirToDir(movement_[z]);
+
+   if(!TryMove(step))
```

Comment:
Diagonal movements do not work now in Griefly, and it will be needed to rework a lot in order to add them. BYOND can into diagonal movements, SS13, however, avoids them. Diagonal movements are not neccesary -> do not use them at least now (and more likely never).